geomesa-core/src/main/scala/org/locationtech/geomesa/core/iterators/DensityIterator.scala

```diff
@@ -294,5 +219,12 @@ object DensityIterator extends Logging {
     }
     table
   }
+}
 
+case class DensityIteratorResult(geometry: Geometry,
+                                 densityGrid: SparseMatrix = HashBasedTable.create[Double, Double, Long]()) extends Result {
+  override def addToFeature(featureBuilder: SimpleFeatureBuilder): Unit = {
+    featureBuilder.add(DensityIterator.encodeSparseMatrix(densityGrid))
+    featureBuilder.add(geometry)
```

Comment:
The one hinky thing about this procedure is that we have to return *some* geometry.  Which one is more or less irrelevant.